From 4c1906f3575b7893b1bd6ae6e77332c67991f1f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jaime=20Marqui=CC=81nez=20Ferra=CC=81ndiz?=
 <jaime.marquinez.ferrandiz@fastmail.net>
Date: Sat, 16 Mar 2024 22:06:29 +0100
Subject: [PATCH] Fix test_different_python under pyhon 3.12

setuptools is no longer preinstalled by the venv module.
---
 test_piplicenses.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test_piplicenses.py b/test_piplicenses.py
index 41d09cf..5a704b1 100644
--- a/test_piplicenses.py
+++ b/test_piplicenses.py
@@ -867,7 +867,10 @@ def test_different_python() -> None:
         package_names = sorted(set(p["name"] for p in pkgs))
         print(package_names)
 
-    assert package_names == ["pip", "setuptools"]
+    expected_packages = ["pip"]
+    if sys.version_info < (3, 12, 0):
+        expected_packages.append("setuptools")
+    assert package_names == expected_packages
 
 
 def test_fail_on(monkeypatch) -> None:
-- 
2.44.0

